### PR TITLE
```Search shards``` API tests

### LINKED ISSUE #663: [BUG] Add missing API Spec Tests
### What is the bug?

API specs test coverage using output added in https://github.com/opensearch-project/opensearch-api-specification/pull/697.

- [x] GET /
- [x] HEAD /
- [x] GET /_alias
- [x] PUT /_alias
- [x] GET /_alias/{name}
- [x] HEAD /_alias/{name}
- [x] POST /_alias/{name}
- [x] PUT /_alias/{name}
- [x] POST /_aliases
- [x] POST /_aliases/{name}
- [x] PUT /_aliases/{name}
- [x] GET /_analyze
- [x] POST /_analyze
- [x] POST /_bulk
- [x] PUT /_bulk
- [x] POST /_bulk/stream
- [x] PUT /_bulk/stream
- [x] POST /_cache/clear
- [x] GET /_cat
- [x] GET /_cat/aliases
- [x] GET /_cat/aliases/{name}
- [x] GET /_cat/allocation
- [x] GET /_cat/allocation/{node_id}
- [x] GET /_cat/cluster_manager
- [x] GET /_cat/count
- [x] GET /_cat/count/{index}
- [x] GET /_cat/fielddata
- [x] GET /_cat/fielddata/{fields}
- [x] GET /_cat/health
- [x] GET /_cat/indices
- [x] GET /_cat/indices/{index}
- [x] GET /_cat/master
- [x] GET /_cat/nodeattrs
- [x] GET /_cat/nodes
- [x] GET /_cat/pending_tasks
- [x] GET /_cat/pit_segments
- [x] GET /_cat/pit_segments/_all
- [x] GET /_cat/plugins
- [x] GET /_cat/recovery
- [x] GET /_cat/recovery/{index}
- [x] GET /_cat/repositories
- [x] GET /_cat/segment_replication
- [x] GET /_cat/segment_replication/{index}
- [x] GET /_cat/segments
- [x] GET /_cat/segments/{index}
- [x] GET /_cat/shards
- [x] GET /_cat/shards/{index}
- [x] GET /_cat/snapshots
- [x] GET /_cat/snapshots/{repository}
- [x] GET /_cat/tasks
- [x] GET /_cat/templates
- [x] GET /_cat/templates/{name}
- [x] GET /_cat/thread_pool
- [x] GET /_cat/thread_pool/{thread_pool_patterns}
- [x] GET /_cluster/allocation/explain
- [x] POST /_cluster/allocation/explain
- [x] DELETE /_cluster/decommission/awareness
- [x] GET /_cluster/decommission/awareness/{awareness_attribute_name}/_status
- [x] PUT /_cluster/decommission/awareness/{awareness_attribute_name}/{awareness_attribute_value}
- [x] GET /_cluster/health
- [x] GET /_cluster/health/{index}
- [x] GET /_cluster/nodes/hot_threads
- [x] GET /_cluster/nodes/{node_id}/hot_threads
- [x] GET /_cluster/pending_tasks
- [x] POST /_cluster/reroute
- [x] DELETE /_cluster/routing/awareness/weights
- [x] GET /_cluster/routing/awareness/{attribute}/weights
- [x] PUT /_cluster/routing/awareness/{attribute}/weights
- [x] GET /_cluster/settings
- [x] PUT /_cluster/settings
- [x] GET /_cluster/state
- [x] GET /_cluster/state/{metric}
- [x] GET /_cluster/state/{metric}/{index}
- [x] GET /_cluster/stats
- [x] GET /_cluster/stats/nodes/{node_id}
- [x] GET /_cluster/stats/{metric}/nodes/{node_id}
- [x] GET /_cluster/stats/{metric}/{index_metric}/nodes/{node_id}
- [x] DELETE /_cluster/voting_config_exclusions
- [x] POST /_cluster/voting_config_exclusions
- [x] GET /_component_template
- [x] DELETE /_component_template/{name}
- [x] GET /_component_template/{name}
- [x] HEAD /_component_template/{name}
- [x] POST /_component_template/{name}
- [x] PUT /_component_template/{name}
- [x] GET /_count
- [x] POST /_count
- [x] GET /_dangling
- [x] DELETE /_dangling/{index_uuid}
- [x] POST /_dangling/{index_uuid}
- [x] GET /_data_stream
- [x] GET /_data_stream/_stats
- [x] DELETE /_data_stream/{name}
- [x] GET /_data_stream/{name}
- [x] PUT /_data_stream/{name}
- [x] GET /_data_stream/{name}/_stats
- [x] POST /_delete_by_query/{task_id}/_rethrottle
- [x] GET /_field_caps
- [x] POST /_field_caps
- [x] GET /_flush
- [x] POST /_flush
- [x] POST /_forcemerge
- [x] GET /_index_template
- [x] POST /_index_template/_simulate
- [x] POST /_index_template/_simulate/{name}
- [x] POST /_index_template/_simulate_index/{name}
- [x] DELETE /_index_template/{name}
- [x] GET /_index_template/{name}
- [x] HEAD /_index_template/{name}
- [x] POST /_index_template/{name}
- [x] PUT /_index_template/{name}
- [x] GET /_ingest/pipeline
- [x] GET /_ingest/pipeline/_simulate
- [x] POST /_ingest/pipeline/_simulate
- [x] DELETE /_ingest/pipeline/{id}
- [x] GET /_ingest/pipeline/{id}
- [x] PUT /_ingest/pipeline/{id}
- [x] GET /_ingest/pipeline/{id}/_simulate
- [x] POST /_ingest/pipeline/{id}/_simulate
- [x] GET /_ingest/processor/grok
- [x] GET /_insights/top_queries
- [x] GET /_list
- [x] GET /_list/indices
- [x] GET /_list/indices/{index}
- [x] GET /_list/shards
- [x] GET /_list/shards/{index}
- [x] GET /_mapping
- [x] GET /_mapping/field/{fields}
- [x] GET /_mget
- [x] POST /_mget
- [x] GET /_msearch
- [x] POST /_msearch
- [x] GET /_msearch/template
- [x] POST /_msearch/template
- [x] GET /_mtermvectors
- [x] POST /_mtermvectors
- [x] GET /_nodes
- [x] GET /_nodes/hot_threads
- [x] POST /_nodes/reload_secure_settings
- [x] GET /_nodes/stats
- [x] GET /_nodes/stats/{metric}
- [x] GET /_nodes/stats/{metric}/{index_metric}
- [x] GET /_nodes/usage
- [x] GET /_nodes/usage/{metric}
- [x] GET /_nodes/{node_id_or_metric}
- [x] GET /_nodes/{node_id}/hot_threads
- [x] POST /_nodes/{node_id}/reload_secure_settings
- [x] GET /_nodes/{node_id}/stats
- [x] GET /_nodes/{node_id}/stats/{metric}
- [x] GET /_nodes/{node_id}/stats/{metric}/{index_metric}
- [x] GET /_nodes/{node_id}/usage
- [x] GET /_nodes/{node_id}/usage/{metric}
- [x] GET /_nodes/{node_id}/{metric}
- [x] GET /_opendistro/_security/sslinfo
- [x] POST /_plugins/_asynchronous_search
- [x] GET /_plugins/_asynchronous_search/stats
- [x] DELETE /_plugins/_asynchronous_search/{id}
- [x] GET /_plugins/_asynchronous_search/{id}
- [x] POST /_plugins/_flow_framework/workflow
- [x] GET /_plugins/_flow_framework/workflow/_search
- [x] POST /_plugins/_flow_framework/workflow/_search
- [x] GET /_plugins/_flow_framework/workflow/_steps
- [x] GET /_plugins/_flow_framework/workflow/state/_search
- [x] POST /_plugins/_flow_framework/workflow/state/_search
- [x] DELETE /_plugins/_flow_framework/workflow/{workflow_id}
- [x] GET /_plugins/_flow_framework/workflow/{workflow_id}
- [x] PUT /_plugins/_flow_framework/workflow/{workflow_id}
- [x] POST /_plugins/_flow_framework/workflow/{workflow_id}/_deprovision
- [x] POST /_plugins/_flow_framework/workflow/{workflow_id}/_provision
- [x] GET /_plugins/_flow_framework/workflow/{workflow_id}/_status
- [x] POST /_plugins/_ism/add
- [x] POST /_plugins/_ism/add/{index}
- [x] POST /_plugins/_ism/change_policy
- [x] POST /_plugins/_ism/change_policy/{index}
- [x] GET /_plugins/_ism/explain
- [x] POST /_plugins/_ism/explain
- [x] GET /_plugins/_ism/explain/{index}
- [x] POST /_plugins/_ism/explain/{index}
- [x] GET /_plugins/_ism/policies
- [x] PUT /_plugins/_ism/policies
- [x] DELETE /_plugins/_ism/policies/{policy_id}
- [x] GET /_plugins/_ism/policies/{policy_id}
- [x] HEAD /_plugins/_ism/policies/{policy_id}
- [x] PUT /_plugins/_ism/policies/{policy_id}
- [x] POST /_plugins/_ism/remove
- [x] POST /_plugins/_ism/remove/{index}
- [x] POST /_plugins/_ism/retry
- [x] POST /_plugins/_ism/retry/{index}
- [x] GET /_plugins/_knn/models/_search
- [x] POST /_plugins/_knn/models/_search
- [x] POST /_plugins/_knn/models/_train
- [x] DELETE /_plugins/_knn/models/{model_id}
- [x] GET /_plugins/_knn/models/{model_id}
- [x] POST /_plugins/_knn/models/{model_id}/_train
- [x] GET /_plugins/_knn/stats
- [x] GET /_plugins/_knn/stats/{stat}
- [x] GET /_plugins/_knn/warmup/{index}
- [x] GET /_plugins/_knn/{node_id}/stats
- [x] GET /_plugins/_knn/{node_id}/stats/{stat}
- [x] POST /_plugins/_ml/agents/_register
- [x] DELETE /_plugins/_ml/agents/{agent_id}
- [x] POST /_plugins/_ml/connectors/_create
- [x] DELETE /_plugins/_ml/connectors/{connector_id}
- [x] POST /_plugins/_ml/model_groups/_register
- [x] DELETE /_plugins/_ml/model_groups/{model_group_id}
- [x] GET /_plugins/_ml/model_groups/{model_group_id}
- [x] POST /_plugins/_ml/models/_register
- [x] GET /_plugins/_ml/models/_search
- [x] DELETE /_plugins/_ml/models/{model_id}
- [x] POST /_plugins/_ml/models/{model_id}/_deploy
- [x] POST /_plugins/_ml/models/{model_id}/_undeploy
- [x] DELETE /_plugins/_ml/tasks/{task_id}
- [x] GET /_plugins/_ml/tasks/{task_id}
- [x] GET /_plugins/_notifications/channels
- [x] DELETE /_plugins/_notifications/configs
- [x] GET /_plugins/_notifications/configs
- [x] POST /_plugins/_notifications/configs
- [x] DELETE /_plugins/_notifications/configs/{config_id}
- [x] GET /_plugins/_notifications/configs/{config_id}
- [x] PUT /_plugins/_notifications/configs/{config_id}
- [x] GET /_plugins/_notifications/feature/test/{config_id}
- [x] POST /_plugins/_notifications/feature/test/{config_id}
- [x] GET /_plugins/_notifications/features
- [x] GET /_plugins/_observability/_local/stats
- [x] DELETE /_plugins/_observability/object
- [x] GET /_plugins/_observability/object
- [x] POST /_plugins/_observability/object
- [x] DELETE /_plugins/_observability/object/{object_id}
- [x] GET /_plugins/_observability/object/{object_id}
- [x] PUT /_plugins/_observability/object/{object_id}
- [x] POST /_plugins/_ppl
- [x] POST /_plugins/_ppl/_explain
- [x] GET /_plugins/_ppl/stats
- [x] POST /_plugins/_ppl/stats
- [x] GET /_plugins/_query/_datasources
- [x] POST /_plugins/_query/_datasources
- [x] PUT /_plugins/_query/_datasources
- [x] DELETE /_plugins/_query/_datasources/{datasource_name}
- [x] GET /_plugins/_query/_datasources/{datasource_name}
- [x] PUT /_plugins/_query/settings
- [x] POST /_plugins/_refresh_search_analyzers/{index}
- [x] DELETE /_plugins/_replication/_autofollow
- [x] POST /_plugins/_replication/_autofollow
- [x] GET /_plugins/_replication/autofollow_stats
- [x] GET /_plugins/_replication/follower_stats
- [x] GET /_plugins/_replication/leader_stats
- [x] POST /_plugins/_replication/{index}/_pause
- [x] POST /_plugins/_replication/{index}/_resume
- [x] PUT /_plugins/_replication/{index}/_start
- [x] GET /_plugins/_replication/{index}/_status
- [x] POST /_plugins/_replication/{index}/_stop
- [x] PUT /_plugins/_replication/{index}/_update
- [x] DELETE /_plugins/_rollup/jobs/{id}
- [x] GET /_plugins/_rollup/jobs/{id}
- [x] PUT /_plugins/_rollup/jobs/{id}
- [x] GET /_plugins/_rollup/jobs/{id}/_explain
- [x] POST /_plugins/_rollup/jobs/{id}/_start
- [x] POST /_plugins/_rollup/jobs/{id}/_stop
- [x] GET /_plugins/_security/api/_upgrade_check
- [x] POST /_plugins/_security/api/_upgrade_perform
- [x] GET /_plugins/_security/api/account
- [x] PUT /_plugins/_security/api/account
- [x] GET /_plugins/_security/api/actiongroups
- [x] PATCH /_plugins/_security/api/actiongroups
- [x] DELETE /_plugins/_security/api/actiongroups/{action_group}
- [x] GET /_plugins/_security/api/actiongroups/{action_group}
- [x] PATCH /_plugins/_security/api/actiongroups/{action_group}
- [x] PUT /_plugins/_security/api/actiongroups/{action_group}
- [x] GET /_plugins/_security/api/allowlist
- [x] PATCH /_plugins/_security/api/allowlist
- [x] PUT /_plugins/_security/api/allowlist
- [x] GET /_plugins/_security/api/audit
- [x] PATCH /_plugins/_security/api/audit
- [x] PUT /_plugins/_security/api/audit/config
- [x] POST /_plugins/_security/api/authtoken
- [x] DELETE /_plugins/_security/api/cache
- [x] GET /_plugins/_security/api/cache
- [x] POST /_plugins/_security/api/cache
- [x] PUT /_plugins/_security/api/cache
- [x] GET /_plugins/_security/api/certificates
- [x] GET /_plugins/_security/api/certificates/{node_id}
- [x] POST /_plugins/_security/api/generateonbehalfoftoken
- [x] GET /_plugins/_security/api/internalusers
- [x] PATCH /_plugins/_security/api/internalusers
- [x] DELETE /_plugins/_security/api/internalusers/{username}
- [x] GET /_plugins/_security/api/internalusers/{username}
- [x] PATCH /_plugins/_security/api/internalusers/{username}
- [x] PUT /_plugins/_security/api/internalusers/{username}
- [x] POST /_plugins/_security/api/internalusers/{username}/authtoken
- [x] POST /_plugins/_security/api/migrate
- [x] GET /_plugins/_security/api/nodesdn
- [x] PATCH /_plugins/_security/api/nodesdn
- [x] DELETE /_plugins/_security/api/nodesdn/{cluster_name}
- [x] GET /_plugins/_security/api/nodesdn/{cluster_name}
- [x] PATCH /_plugins/_security/api/nodesdn/{cluster_name}
- [x] PUT /_plugins/_security/api/nodesdn/{cluster_name}
- [x] GET /_plugins/_security/api/permissionsinfo
- [x] GET /_plugins/_security/api/roles
- [x] PATCH /_plugins/_security/api/roles
- [x] DELETE /_plugins/_security/api/roles/{role}
- [x] GET /_plugins/_security/api/roles/{role}
- [x] PATCH /_plugins/_security/api/roles/{role}
- [x] PUT /_plugins/_security/api/roles/{role}
- [x] GET /_plugins/_security/api/rolesmapping
- [x] PATCH /_plugins/_security/api/rolesmapping
- [x] DELETE /_plugins/_security/api/rolesmapping/{role}
- [x] GET /_plugins/_security/api/rolesmapping/{role}
- [x] PATCH /_plugins/_security/api/rolesmapping/{role}
- [x] PUT /_plugins/_security/api/rolesmapping/{role}
- [x] GET /_plugins/_security/api/securityconfig
- [x] PATCH /_plugins/_security/api/securityconfig
- [x] PUT /_plugins/_security/api/securityconfig/config
- [x] GET /_plugins/_security/api/ssl/certs
- [x] PUT /_plugins/_security/api/ssl/http/reloadcerts
- [x] PUT /_plugins/_security/api/ssl/transport/reloadcerts
- [x] GET /_plugins/_security/api/tenancy/config
- [x] PUT /_plugins/_security/api/tenancy/config
- [x] GET /_plugins/_security/api/tenants
- [x] PATCH /_plugins/_security/api/tenants
- [x] DELETE /_plugins/_security/api/tenants/{tenant}
- [x] GET /_plugins/_security/api/tenants/{tenant}
- [x] PATCH /_plugins/_security/api/tenants/{tenant}
- [x] PUT /_plugins/_security/api/tenants/{tenant}
- [x] GET /_plugins/_security/api/user
- [x] DELETE /_plugins/_security/api/user/{username}
- [x] GET /_plugins/_security/api/user/{username}
- [x] PUT /_plugins/_security/api/user/{username}
- [x] POST /_plugins/_security/api/user/{username}/authtoken
- [x] GET /_plugins/_security/api/validate
- [x] GET /_plugins/_security/authinfo
- [x] POST /_plugins/_security/authinfo
- [x] GET /_plugins/_security/dashboardsinfo
- [x] POST /_plugins/_security/dashboardsinfo
- [x] GET /_plugins/_security/health
- [x] POST /_plugins/_security/health
- [x] GET /_plugins/_security/tenantinfo
- [x] POST /_plugins/_security/tenantinfo
- [x] GET /_plugins/_security/whoami
- [x] POST /_plugins/_security/whoami
- [x] GET /_plugins/_security/whoamiprotected
- [x] GET /_plugins/_sm/policies
- [x] DELETE /_plugins/_sm/policies/{policy_name}
- [x] GET /_plugins/_sm/policies/{policy_name}
- [x] POST /_plugins/_sm/policies/{policy_name}
- [x] PUT /_plugins/_sm/policies/{policy_name}
- [x] GET /_plugins/_sm/policies/{policy_name}/_explain
- [x] POST /_plugins/_sm/policies/{policy_name}/_start
- [x] POST /_plugins/_sm/policies/{policy_name}/_stop
- [x] POST /_plugins/_sql
- [x] POST /_plugins/_sql/_explain
- [x] POST /_plugins/_sql/close
- [x] GET /_plugins/_sql/stats
- [x] POST /_plugins/_sql/stats
- [x] GET /_plugins/_transform
- [x] POST /_plugins/_transform/_preview
- [x] DELETE /_plugins/_transform/{id}
- [x] GET /_plugins/_transform/{id}
- [x] PUT /_plugins/_transform/{id}
- [x] GET /_plugins/_transform/{id}/_explain
- [x] POST /_plugins/_transform/{id}/_start
- [x] POST /_plugins/_transform/{id}/_stop
- [x] GET /_rank_eval
- [x] POST /_rank_eval
- [x] GET /_recovery
- [x] GET /_refresh
- [x] POST /_refresh
- [x] POST /_reindex
- [x] POST /_reindex/{task_id}/_rethrottle
- [x] GET /_remote/info
- [x] POST /_remotestore/_restore
- [x] GET /_render/template
- [x] POST /_render/template
- [x] GET /_render/template/{id}
- [x] POST /_render/template/{id}
- [x] GET /_resolve/index/{name}
- [x] GET /_script_context
- [x] GET /_script_language
- [x] GET /_scripts/painless/_execute
- [x] POST /_scripts/painless/_execute
- [x] DELETE /_scripts/{id}
- [x] GET /_scripts/{id}
- [x] POST /_scripts/{id}
- [x] PUT /_scripts/{id}
- [x] POST /_scripts/{id}/{context}
- [x] PUT /_scripts/{id}/{context}
- [x] GET /_search
- [x] POST /_search
- [x] GET /_search/pipeline
- [x] DELETE /_search/pipeline/{id}
- [x] GET /_search/pipeline/{id}
- [x] PUT /_search/pipeline/{id}
- [x] DELETE /_search/point_in_time
- [x] DELETE /_search/point_in_time/_all
- [x] GET /_search/point_in_time/_all
- [x] DELETE /_search/scroll
- [x] GET /_search/scroll
- [x] POST /_search/scroll
- [x] DELETE /_search/scroll/{scroll_id}
- [x] GET /_search/scroll/{scroll_id}
- [x] POST /_search/scroll/{scroll_id}
- [x] GET /_search/template
- [x] POST /_search/template
- [x] GET /_search_shards
- [x] POST /_search_shards
- [x] GET /_segments
- [x] GET /_settings
- [x] PUT /_settings
- [x] GET /_settings/{name}
- [x] GET /_shard_stores
- [x] GET /_snapshot
- [x] GET /_snapshot/_status
- [x] DELETE /_snapshot/{repository}
- [x] GET /_snapshot/{repository}
- [x] POST /_snapshot/{repository}
- [x] PUT /_snapshot/{repository}
- [x] POST /_snapshot/{repository}/_cleanup
- [x] GET /_snapshot/{repository}/_status
- [x] POST /_snapshot/{repository}/_verify
- [x] DELETE /_snapshot/{repository}/{snapshot}
- [x] GET /_snapshot/{repository}/{snapshot}
- [x] POST /_snapshot/{repository}/{snapshot}
- [x] PUT /_snapshot/{repository}/{snapshot}
- [x] PUT /_snapshot/{repository}/{snapshot}/_clone/{target_snapshot}
- [x] POST /_snapshot/{repository}/{snapshot}/_restore
- [x] GET /_snapshot/{repository}/{snapshot}/_status
- [x] GET /_stats
- [x] GET /_stats/{metric}
- [x] GET /_tasks
- [x] POST /_tasks/_cancel
- [x] GET /_tasks/{task_id}
- [x] POST /_tasks/{task_id}/_cancel
- [x] GET /_template
- [x] DELETE /_template/{name}
- [x] GET /_template/{name}
- [x] HEAD /_template/{name}
- [x] POST /_template/{name}
- [x] PUT /_template/{name}
- [x] POST /_update_by_query/{task_id}/_rethrottle
- [x] GET /_upgrade
- [x] POST /_upgrade
- [x] GET /_validate/query
- [x] POST /_validate/query
- [x] GET /_wlm/query_group
- [x] PUT /_wlm/query_group
- [x] DELETE /_wlm/query_group/{name}
- [x] GET /_wlm/query_group/{name}
- [x] PUT /_wlm/query_group/{name}
- [x] POST /{alias}/_rollover
- [x] POST /{alias}/_rollover/{new_index}
- [x] DELETE /{index}
- [x] GET /{index}
- [x] HEAD /{index}
- [x] PUT /{index}
- [x] GET /{index}/_alias
- [x] PUT /{index}/_alias
- [x] DELETE /{index}/_alias/{name}
- [x] GET /{index}/_alias/{name}
- [x] HEAD /{index}/_alias/{name}
- [x] POST /{index}/_alias/{name}
- [x] PUT /{index}/_alias/{name}
- [x] PUT /{index}/_aliases
- [x] DELETE /{index}/_aliases/{name}
- [x] POST /{index}/_aliases/{name}
- [x] PUT /{index}/_aliases/{name}
- [x] GET /{index}/_analyze
- [x] POST /{index}/_analyze
- [x] PUT /{index}/_block/{block}
- [x] POST /{index}/_bulk
- [x] PUT /{index}/_bulk
- [x] POST /{index}/_bulk/stream
- [x] PUT /{index}/_bulk/stream
- [x] POST /{index}/_cache/clear
- [x] POST /{index}/_clone/{target}
- [x] PUT /{index}/_clone/{target}
- [x] POST /{index}/_close
- [x] GET /{index}/_count
- [x] POST /{index}/_count
- [x] POST /{index}/_create/{id}
- [x] PUT /{index}/_create/{id}
- [x] POST /{index}/_delete_by_query
- [x] POST /{index}/_doc
- [x] DELETE /{index}/_doc/{id}
- [x] GET /{index}/_doc/{id}
- [x] HEAD /{index}/_doc/{id}
- [x] POST /{index}/_doc/{id}
- [x] PUT /{index}/_doc/{id}
- [x] GET /{index}/_explain/{id}
- [x] POST /{index}/_explain/{id}
- [x] GET /{index}/_field_caps
- [x] POST /{index}/_field_caps
- [x] GET /{index}/_flush
- [x] POST /{index}/_flush
- [x] POST /{index}/_forcemerge
- [x] GET /{index}/_mapping
- [x] POST /{index}/_mapping
- [x] PUT /{index}/_mapping
- [x] GET /{index}/_mapping/field/{fields}
- [x] GET /{index}/_mget
- [x] POST /{index}/_mget
- [x] GET /{index}/_msearch
- [x] POST /{index}/_msearch
- [x] GET /{index}/_msearch/template
- [x] POST /{index}/_msearch/template
- [x] GET /{index}/_mtermvectors
- [x] POST /{index}/_mtermvectors
- [x] POST /{index}/_open
- [x] GET /{index}/_rank_eval
- [x] POST /{index}/_rank_eval
- [x] GET /{index}/_recovery
- [x] GET /{index}/_refresh
- [x] POST /{index}/_refresh
- [x] GET /{index}/_search
- [x] POST /{index}/_search
- [x] POST /{index}/_search/point_in_time
- [x] GET /{index}/_search/template
- [x] POST /{index}/_search/template
- [x] GET /{index}/_search_shards
- [x] POST /{index}/_search_shards
- [x] GET /{index}/_segments
- [x] GET /{index}/_settings
- [x] PUT /{index}/_settings
- [x] GET /{index}/_settings/{name}
- [x] GET /{index}/_shard_stores
- [x] POST /{index}/_shrink/{target}
- [x] PUT /{index}/_shrink/{target}
- [x] GET /{index}/_source/{id}
- [x] HEAD /{index}/_source/{id}
- [x] POST /{index}/_split/{target}
- [x] PUT /{index}/_split/{target}
- [x] GET /{index}/_stats
- [x] GET /{index}/_stats/{metric}
- [x] GET /{index}/_termvectors
- [x] POST /{index}/_termvectors
- [x] GET /{index}/_termvectors/{id}
- [x] POST /{index}/_termvectors/{id}
- [x] POST /{index}/_update/{id}
- [x] POST /{index}/_update_by_query
- [x] GET /{index}/_upgrade
- [x] POST /{index}/_upgrade
- [x] GET /{index}/_validate/query
- [x] POST /{index}/_validate/query

### What is the expected behavior?

100% test coverage.

### Do you have any additional context?

1. Read https://github.com/opensearch-project/opensearch-api-specification/blob/main/TESTING_GUIDE.md.
2. Pick a test from the list to add. Every PR will have a step called "merge test coverage" that will show missing ones, [example](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/11785290593/job/32826305084?pr=662).

Screenshot 2024-11-11 at 3 01 48 PM

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
 GET /{index}/_search_shards
 POST /{index}/_search_shards
 GET /_search_shards
 POST /_search_shards

### Issues Resolved
#663 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/_core.yaml [hunk 1]
```diff
@@ -2690,6 +2698,46 @@ components:
...
+          schema:
+            type: object
+            properties:
+              query:
+                description: Defines the search query to filter shards. Supports any query that is valid for the `_search` endpoint.
+                type: object
+                additionalProperties:
+                  type: object
+              indices_boost:
+                description: Allows increasing the relevance of specific indices in the search.
```
Comment (dblock) [inline spec/namespaces/_core.yaml:2713]:
```suggestion
                description: Allows increasing the relevance of specific indexes in the search.
```

**Review by dblock — APPROVED** [review]
For next time, "indexes" is what the documentation uses for indices, https://github.com/opensearch-project/documentation-website/issues/8696

**Comment by Xtansia** [comment]
@Tokesh Do you happen to remember what source you used that indicated `/_search_shards` supported a request body?

**Comment by Tokesh** [comment]
> @Tokesh Do you happen to remember what source you used that indicated `/_search_shards` supported a request body?

Hi! I wrote about 100+ tests and wrote this a month ago, to be honest I don't remember about it anymore. If you need, let me know, I'll figure it out and give you an answer

**Comment by Xtansia** [comment]
> > @Tokesh Do you happen to remember what source you used that indicated `/_search_shards` supported a request body?
> 
> Hi! I wrote about 100+ tests and wrote this a month ago, to be honest I don't remember about it anymore. If you need, let me know, I'll figure it out and give you an answer

No problem, not important enough to dig into, was just curious if maybe there was incorrect documentation or something floating around

**Pull request merged**